misc/compiler-plugins/modules/configurable-schema-generator/src/test/java/org/ballerinalang/configschemagenerator/ConfigSchemaGenTaskTest.java

```diff
@@ -0,0 +1,122 @@
+
+import io.ballerina.projects.BuildOptions;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.TomlDocument;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.projects.directory.SingleFileProject;
+import io.ballerina.projects.plugins.CompilerPluginException;
+import io.ballerina.projects.util.ProjectConstants;
+import io.ballerina.toml.semantic.ast.TomlTableNode;
+import io.ballerina.toml.validator.TomlValidator;
+import io.ballerina.toml.validator.schema.Schema;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static io.ballerina.projects.util.ProjectConstants.CONFIGURATION_TOML;
+import static org.ballerinalang.configschemagenerator.ConfigSchemaGenTask.CONFIG_SCHEMA;
+
+/**
+ * Tests to verify config schema generation task.
+ */
+public class ConfigSchemaGenTaskTest {
+
+    private static final Path RESOURCES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+    private static final String BALLERINA_HOME_KEY = "ballerina.home";
+
+    @Test(dataProvider = "project-data-provider")
+    public void testGeneratedSchema(String projectType, String projectName, boolean isSingleFileProject) {
+        Path projectPath = RESOURCES_DIR.resolve(projectType).resolve(projectName);
+        Project projectInstance = loadBuildProject(projectPath, isSingleFileProject);
+        projectInstance.currentPackage().getCompilation();
+        TomlValidator tomlValidator = new TomlValidator(Schema.from(
+                readConfigJSONSchema(projectPath, isSingleFileProject)));
+        TomlDocument configToml = TomlDocument.from(CONFIGURATION_TOML,
+                readTomlContent(projectPath, isSingleFileProject));
+        tomlValidator.validate(configToml.toml());
+        TomlTableNode tomlAstNode = configToml.toml().rootNode();
+        if (!tomlAstNode.diagnostics().isEmpty()) {
+            String errorMsg = "";
+            for (Diagnostic diagnostic : tomlAstNode.diagnostics()) {
+                errorMsg = errorMsg.concat(diagnostic.message() + "\n");
+            }
+            Assert.fail(errorMsg);
+        }
+    }
+
+    @DataProvider(name = "project-data-provider")
+    public Object[][] dpMethod() {
+        return new Object[][]{{"DefaultModuleProjects", "SimpleTypeConfigs", false},
+                {"DefaultModuleProjects", "ComplexTypeConfigs", false},
+                {"MultiModuleProjects", "SimpleTypeConfigs", false},
+                {"SingleFileProject1", "testconfig.bal", true}};
+    }
+
+    static Project loadBuildProject(Path projectPath, boolean isSingleFileProject) {
+        System.setProperty(BALLERINA_HOME_KEY, System.getenv(BALLERINA_HOME_KEY));
+        BuildOptions buildOptions = BuildOptions.builder().setOffline(true).build();
```

Comment:
We should use the methods in TestUtils to load the project.

Reply:
@azinneera are you referring to https://github.com/ballerina-platform/ballerina-lang/blob/master/project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestUtils.java ?
It's not exported it seems.

Reply:
we can export it